File: java/src/Othello.java
/**
 * This class contains the main function.
 */
public class Othello {
    /**
     * This is the main function, it creates the position, the evaluator, the alpha beta algorithm and print the
     * determined best move to do.
     * @param args args[0] is the string position of the game, args[1] is the time limit.
     */
    public static void main(String[] args) {
        String pos_str = args[0];
        double time_limit = Double.parseDouble(args[1]);
        Position position = new Position(pos_str);
        Evaluator evaluator = new Evaluator(position.getPlayer());
        AlphaBeta algorithm = new AlphaBeta(evaluator, time_limit);
        Move move = algorithm.bestMove(position);
        System.out.println(move);
    }
}

File: python/othello.py
from move import Move
from alpha_beta import AlphaBeta
from position import Position
from counting_evaluator import CountingEvaluator
import sys

if __name__ == '__main__':
    
    pos_str = sys.argv[1]
    time = float(sys.argv[2])

    position = Position(pos_str)
    evaluator = CountingEvaluator(position.get_player())
    algorithm = AlphaBeta(evaluator, time)
    move = algorithm.best_move(position)
    print(move)
    """
    pos_str = 'WEEEEEEEEEEEEEEEEEEEXOEEEEEEXOEEEEEEXOEEEEEEEEEEEEEEEEEEEEEEEEEEE'
    position = Position(pos_str)
    evaluator = CountingEvaluator(position.get_player())
    print(position)
    print(evaluator.evaluate(position))
    move = Move(x=2, y=2)
    position.make_move(move)
    print(position)
    print(evaluator.evaluate(position))
    """
    
